### ISSUE TITLE
Select renders an input with blank value causing screenreader accessibility concerns

### ISSUE THREAD

**Issue by shivaraja-tte** [ant-design/ant-design#52962]
Component - `Select`

`input` element rendered by select component has empty value attribute causing screen reader accessibility issue.

Expected: `value` attribute should have the label of selected value so that the screenreader reads out the selected value when the focus is on select box. In case of below screenshot `input` should have `value="Jack"`

Refer to below screenshot from the antd document page:

<img width="1183" alt="Image" src="https://github.com/user-attachments/assets/2f74cd18-2740-404d-bdfe-8b1b36fa5365" />

Screenshot showing how screenreader reads the select box:
Screenreader should also read currently selected value when select box is in focus.

<img width="1112" alt="Image" src="https://github.com/user-attachments/assets/2402dd68-28f6-4e0c-90ec-0e94525e1c46" />

And unfortunately unable to modify `value` attribute using `ref`.
